src/org/zaproxy/zap/extension/bugtracker/RaiseSemiAutoIssueDialog.java

```diff
@@ -0,0 +1,120 @@
+ * See the License for the specific language governing permissions and 
+ * limitations under the License. 
+ */
+package org.zaproxy.zap.extension.bugtracker;
+
+import java.awt.Dimension;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.awt.Frame;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.view.View;
+import org.zaproxy.zap.view.StandardFieldsDialog;
+
+public class RaiseSemiAutoIssueDialog extends StandardFieldsDialog {
+
+    private static final long serialVersionUID = -3223449799557586758L;
+
+    private ExtensionBugTracker extension = null;
+    private List<BugTracker> bugTrackers = null;
+    
+    private Set<Alert> alerts = null;
+    private String FIELD_TRACKER_LIST = "bugtracker.trackers.list";
+
+    public RaiseSemiAutoIssueDialog(ExtensionBugTracker ext, Frame owner, Dimension dim){
+        super(owner, "bugtracker.dialog.semi.title", dim);
+        this.extension = ext;
+        this.alerts = ext.alerts;
+        bugTrackers = extension.getBugTrackers();
+        initialize();
+    }
+
+    public void setAlert(Set<Alert> alerts) {
+        this.alerts = alerts;
+        initialize();
+    }
+
+    /**
+     * This method initializes this
+     */
+    private void initialize() {
+        this.removeAllFields();
+        this.setTitle(Constant.messages.getString("bugtracker.popup.issue.semi"));
+        addTrackerList(bugTrackers.get(0).getName());
+        updateTrackerFields();
+        
+    }
+
+    public void addTrackerList(String value) {
+        List<String> trackerNames = new ArrayList<String>();
+        for(BugTracker bugTracker: bugTrackers) {
+            trackerNames.add(bugTracker.getName());
+        }
+        this.addComboField(FIELD_TRACKER_LIST, trackerNames, value);
+        for(BugTracker bugTracker: bugTrackers) {
+            bugTracker.setDialog(this);
+            bugTracker.setDetails(alerts);
+        }
+        this.addFieldListener(FIELD_TRACKER_LIST, new ActionListener() {
+            @Override
+            public void actionPerformed(ActionEvent e) {
+                updateTrackerFields();
+            }
+        });
+    }
+
+    public void updateTrackerFields() {
+        String currentItem = this.getStringValue(FIELD_TRACKER_LIST);
+        for(BugTracker bugTracker: bugTrackers) {
+            if(bugTracker.getName().equals(currentItem)) {
+                this.removeAllFields();
+                addTrackerList(bugTracker.getName());
+                bugTracker.createDialogs();
+                revalidate();
+                repaint();
+            }
+        }
+    }
+
+    @Override
+    public String validateFields() {
+        String currentItem = this.getStringValue(FIELD_TRACKER_LIST);
+        for(BugTracker bugTracker: bugTrackers) {
```

Comment:
This is definitely in the wrong place. The 'validateFields' method is just for checking that all of the input is valid. So this would be a good place to check that the user has supplied a password ;) You should move the code for raising the alerts into the save() method.